internal/exec/describe_affected.go

```diff
@@ -111,11 +121,95 @@ func NewDescribeAffectedExec(
+	var atmosConfig schema.AtmosConfiguration
+	if info, err := ProcessCommandLineArgs("", cmd, args, nil); err != nil {
+		return DescribeAffectedCmdArgs{}, err
+	} else if atmosConfig, err = cfg.InitCliConfig(info, true); err != nil {
+		return DescribeAffectedCmdArgs{}, err
+	}
+	if err := ValidateStacks(atmosConfig); err != nil {
+		return DescribeAffectedCmdArgs{}, err
+	}
+	// Process flags
+	flags := cmd.Flags()
+
+	result := DescribeAffectedCmdArgs{
+		CLIConfig: &atmosConfig,
+	}
+	SetDescribeAffectedFlagValueInCliArgs(flags, &result)
+
+	if result.Format != "yaml" && result.Format != "json" {
+		return DescribeAffectedCmdArgs{}, ErrInvalidFormat
+	}
+	if result.RepoPath != "" && (result.Ref != "" || result.SHA != "" || result.SSHKeyPath != "" || result.SSHKeyPassword != "") {
+		return DescribeAffectedCmdArgs{}, ErrRepoPathConflict
+	}
+
+	return result, nil
+}
+
+// SetDescribeAffectedFlagValueInCliArgs sets the flag values in CLI arguments.
+func SetDescribeAffectedFlagValueInCliArgs(flags *pflag.FlagSet, describe *DescribeAffectedCmdArgs) {
+	flagsKeyValue := map[string]any{
+		"ref":                            &describe.Ref,
+		"sha":                            &describe.SHA,
+		"repo-path":                      &describe.RepoPath,
+		"ssh-key":                        &describe.SSHKeyPath,
+		"ssh-key-password":               &describe.SSHKeyPassword,
+		"include-spacelift-admin-stacks": &describe.IncludeSpaceliftAdminStacks,
+		"include-dependents":             &describe.IncludeDependents,
+		"include-settings":               &describe.IncludeSettings,
+		"upload":                         &describe.Upload,
+		"clone-target-ref":               &describe.CloneTargetRef,
+		"process-templates":              &describe.ProcessTemplates,
+		"process-functions":              &describe.ProcessYamlFunctions,
+		"skip":                           &describe.Skip,
+		"pager":                          &describe.CLIConfig.Settings.Terminal.Pager,
+		"stack":                          &describe.Stack,
+		"format":                         &describe.Format,
+		"file":                           &describe.OutputFile,
+		"query":                          &describe.Query,
+		"verbose":                        &describe.Verbose,
+	}
+
+	var err error
+	for k := range flagsKeyValue {
+		if !flags.Changed(k) {
+			continue
+		}
+		switch v := flagsKeyValue[k].(type) {
+		case *string:
+			*v, err = flags.GetString(k)
+		case *bool:
+			*v, err = flags.GetBool(k)
+		case *[]string:
+			*v, err = flags.GetStringSlice(k)
+		default:
+			panic(fmt.Sprintf("unsupported type %T for flag %s", v, k))
+		}
+		if err != nil {
+			u.PrintErrorMarkdownAndExit("", err, "")
+		}
+	}
+	// When uploading, always include dependents and settings for all affected components
+	if describe.Upload {
+		describe.IncludeDependents = true
+		describe.IncludeSettings = true
+	}
+	if describe.Format == "" {
+		describe.Format = "json"
```

Comment:
It’s weird to me that when we’re uploading, we allow a different format. Atmos pro always expects JSON so we cannot upload in YAMl.